community/kernel/src/main/java/org/neo4j/kernel/impl/api/state/TxState.java

```diff
@@ -1334,6 +1345,45 @@ public boolean hasDataChanges()
         return hasDataChanges;
     }
 
+    /**
+     * Release all underlying resources. The instance must not be used after calling this method .
+     */
+    public void release()
```

Comment:
invoking combinations of close and releases here looks strange

Reply:
Because `close()` is a really bad name for a method releasing resources, but I didn't want to rename it everywhere. Should we do it now?

Reply:
let's have a quick talk about this release call